dd-sdk-android/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt

```diff
@@ -43,6 +52,19 @@ internal class DatadogExceptionHandler(
         )
         writer.write(log)
 
+        contextRef.get()?.let {
```

Comment:
I think this function is getting too big, can you extract some methods from it please.

Reply:
Sure